distsys/mpcalctx.go

```diff
@@ -540,14 +540,19 @@ func (ctx *MPCalContext) Run() error {
 		case nil: // everything is fine; carry on
 		case ErrCriticalSectionAborted:
 			ctx.abort()
-			err = nil
```

Comment:
Honestly I preferred this where it was, logically speaking. Here, it shows the one case that actually requires setting `err = nil`, vs it being nil already, or the function immediately returning.

Linter comment is a good idea though.

Reply:
Got your point, I'll move it back.

Reply:
`ineffassign` was pointing out that `err` is reassigned to before being tested again, so the actual value after the select statement doesn't matter.

Reply:
Sure, I figured it would be something like that. What I mean by my comment is that, while one could carefully analyse the entire for loop and figure that out, that redundant line being there makes me feel better, in terms of signifying that it is not intended for err's non-nil value to be re-used (e.g if someone adds more code later between the select and the next assignment).

At a stretch, putting a comment to the same effect could work, but I figured it would be simplest to just explicitly destroy the reference, and have the restriction "enforced at runtime", if you will.